source/common/upstream/cluster_manager_impl.h

```diff
@@ -361,14 +367,38 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
   // This map is ordered so that config dumping is consistent.
   typedef std::map<std::string, ClusterDataPtr> ClusterMap;
 
+  struct PendingUpdates {
+    Event::TimerPtr timer_;
+    bool timer_enabled_{};
```

Comment:
In general would prefer this was part of the timer interface itself. Can you at least add a TODO here about that?